src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -421,8 +438,27 @@ private async Task<ActionResult<SynchronizePullRequestResult>> SynchronizePullRe
                                 prUrl);
                             await StateManager.RemoveStateAsync(PullRequest);
                             return ActionResult.Create(SynchronizePullRequestResult.Completed, checkPolicyResult.Message);
-                        case MergePolicyCheckResult.NoPolicies:
                         case MergePolicyCheckResult.FailedPolicies:
+                            if (GetType() == typeof(NonBatchedPullRequestActorImplementation))
```

Comment:
It's really, really weird to do type inspection here.  Why not just use polymorphism on the `BatchedPullRequestActorImplementation` class?